src/main/java/decodes/datasource/InputDirectory.java

```diff
@@ -47,9 +48,9 @@ public class InputDirectory
 	extends DataSourceExec
 {
 	/** default constructor */
-	public InputDirectory()
+	public InputDirectory(DataSource ds, Database db)
```

Comment:
With args, I don't think this is considered a default constructor as comment above calls it.

Reply:
There are several other Class that how have args, making the comments about default constructors confusing.

Reply:
Yep. I need to clean that up. I can't remember exactly what I was tracking down but it had to do with the singleton access to the `Database` class and and getting the `DataSource` assigned. but some of those could actually use a default constructor and I didn't even think about the JavaDocs.